### PR TITLE
Sammyk/php docs url resource name

### PR DESCRIPTION
**Pull request by priyanshi-gupta**
### Preview link

https://docs-staging.datadoghq.com/sammyk/php-docs-url-resource-name/tracing/setup/php/#url-resource-names

### CODE REVIEW

File: content/en/tracing/setup/php.md [hunk 1]
```diff
@@ -210,16 +213,62 @@ DD_TRACE_DEBUG=true php -S localhost:8888
...
 | `DD_TRACE_GLOBAL_TAGS`               | ``          | Tags to be set on all spans: e.g.: `key1:value1,key2:value2`                |
+| `DD_TRACE_REPORT_HOSTNAME`           | `false`     | Enable hostname reporting on the root span                                  |
+| `DD_TRACE_RESOURCE_URI_MAPPING`      | `null`      | CSV of URL-to-resource-name mapping rules; e.g., `/foo/*,/bar/$*/baz`; [see "Custom URL-To-Resource Mapping"][16] |
+| `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED` | `false` | Enable URL's as resource names; [see "URL Resource Names"][17]              |
 | `DD_<INTEGRATION>_ANALYTICS_ENABLED` | `false`     | Flag to enable trace analytics for relevant spans in a specific integration |
 
+#### URL Resource Names
+
+When `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED=true`, the URL will be used to form the trace resource name in the format, `<HTTP_REQUEST_METHOD> <NORMALIZED_URL>` with the query string removed from the URL.
+
```
Comment (priyanshi-gupta) [inline content/en/tracing/setup/php.md]: Let's add a line at the end to clarify the usability -  
"...from the URL. This allows better visibility in any custom framework that is not automatically instrumented by normalizing the URLs and grouping together generic endpoints under one resource."

File: content/en/tracing/setup/php.md [hunk 2]
```diff
@@ -210,16 +213,62 @@ DD_TRACE_DEBUG=true php -S localhost:8888
...
 | `DD_TRACE_ANALYTICS_ENABLED`         | `false`     | Flag to enable trace analytics for relevant spans in web integrations       |
 | `DD_TRACE_DEBUG`                     | `false`     | Enable [debug mode][15] for the tracer                                      |
 | `DD_TRACE_ENABLED`                   | `true`      | Enable the tracer globally                                                  |
 | `DD_TRACE_GLOBAL_TAGS`               | ``          | Tags to be set on all spans: e.g.: `key1:value1,key2:value2`                |
+| `DD_TRACE_REPORT_HOSTNAME`           | `false`     | Enable hostname reporting on the root span                                  |
+| `DD_TRACE_RESOURCE_URI_MAPPING`      | `null`      | CSV of URL-to-resource-name mapping rules; e.g., `/foo/*,/bar/$*/baz`; [see "Custom URL-To-Resource Mapping"][16] |
+| `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED` | `false` | Enable URL's as resource names; [see "URL Resource Names"][17]              |
 | `DD_<INTEGRATION>_ANALYTICS_ENABLED` | `false`     | Flag to enable trace analytics for relevant spans in a specific integration |
 
+#### URL Resource Names
```
Comment (priyanshi-gupta) [inline content/en/tracing/setup/php.md]: Can we call the functionality -
"Map resource names to normalized URI"

File: content/en/tracing/setup/php.md [hunk 3]
```diff
@@ -232,6 +232,14 @@ DD_TRACE_DEBUG=true php -S localhost:8888
 
 #### URL Resource Names
 
+<div class="alert alert-warning">
+Enabling this feature <strong>could impact your billing</strong>.
+
+Increasing the number of unique resource names also increases cardinality which is a metric used in billing.
```
Comment (priyanshi-gupta) [inline content/en/tracing/setup/php.md]: It doesn't affect billing directly. We limit after a certain number of resources. I'd suggest dropping this off unless we are expecting any known repercussions. That is, had we auto-instrumented the framework, the same number of resources would have appeared as after normalization. Why do we need to warn them specifically? @labbati thoughts?

Comment (labbati) [reply content/en/tracing/setup/php.md]: This is good to know, so I would definitely rephrase it in a different and less scary way.

> had we auto-instrumented the framework, the same number of resources would have appeared as after normalization. Why do we need to warn them specifically?

When we support a framework we have access to the `route` name in addition to the url and we use the first one as a resource name.

So for example for a Symfony controller you set a route like this:

```
/** @Route("/users/{id}/permissions", name="users_permissions") */
public function getUserPermissions($id) {
   ....
}
```

So here for `/users/123/permissions` we have access to both `/user/{}/permissions` or `user_permissions` names and we can use the one that we like as the resource name. No cardinality issues in this case.

Comment (priyanshi-gupta) [reply content/en/tracing/setup/php.md]: I'll suggest handling it on per-use basis. Let's not use the warning banner to start with. Reason - it won't affect their billing, it'll affect the number of resources that they see after ballooning over a certain limit - which we can treat as a bug and increase the limit or look at the feature again. WDYT?

File: content/en/tracing/setup/php.md [hunk 4]
```diff
@@ -111,16 +111,18 @@ If the web framework that you use is not listed below, you can still see traces
 
 | Module         | Versions           | Support Type    |
 |:---------------|:-------------------|:----------------|
+| CakePHP        | 2.x                | Fully Supported |
 | Laravel        | 4.2, 5.x           | Fully Supported |
 | Lumen          | 5.2+               | Fully Supported |
+| Slim           | 3.x                | Fully Supported |
 | Symfony        | 2.x, 3.3, 3.4, 4.x | Fully Supported |
```
Comment (priyanshi-gupta) [inline content/en/tracing/setup/php.md]: Remove Symfony 2.x

File: content/en/tracing/setup/php.md [hunk 5]
```diff
@@ -210,20 +213,66 @@ DD_TRACE_DEBUG=true php -S localhost:8888
...
 | `DD_TRACE_DEBUG`                     | `false`     | Enable [debug mode][16] for the tracer                                      |
 | `DD_TRACE_ENABLED`                   | `true`      | Enable the tracer globally                                                  |
 | `DD_TRACE_GLOBAL_TAGS`               | ``          | Tags to be set on all spans: e.g.: `key1:value1,key2:value2`                |
+| `DD_TRACE_REPORT_HOSTNAME`           | `false`     | Enable hostname reporting on the root span                                  |
+| `DD_TRACE_RESOURCE_URI_MAPPING`      | `null`      | CSV of URL-to-resource-name mapping rules; e.g., `/foo/*,/bar/$*/baz`; [see "Custom URL-To-Resource Mapping"][16] |
+| `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED` | `false` | Enable URL's as resource names; [see "Map Resource Names To Normalized URI"][17] |
 | `DD_<INTEGRATION>_ANALYTICS_ENABLED` | `false`     | Flag to enable trace analytics for relevant spans in a specific integration |
 
+#### Map Resource Names To Normalized URI
+
```
Comment (apigirl) [inline content/en/tracing/setup/php.md]:
```suggestion

<div class="alert alert-warning">
This functionality is in public beta. Request early access by contacting <a href="/help">Datadog Support</a>. 
</div>
```

File: content/en/tracing/setup/php.md [hunk 6]
```diff
@@ -210,20 +213,69 @@ DD_TRACE_DEBUG=true php -S localhost:8888
...
+<div class="alert alert-warning">
+This functionality is in public beta. Request early access by contacting <a href="/help">Datadog Support</a>. 
+</div>
```
Comment (apigirl) [inline content/en/tracing/setup/php.md]:
```suggestion
</div>

```

File: content/en/tracing/setup/php.md [hunk 7]
```diff
@@ -210,20 +213,70 @@ DD_TRACE_DEBUG=true php -S localhost:8888
...
 | `DD_TRACE_GLOBAL_TAGS`               | ``          | Tags to be set on all spans: e.g.: `key1:value1,key2:value2`                |
+| `DD_TRACE_REPORT_HOSTNAME`           | `false`     | Enable hostname reporting on the root span                                  |
+| `DD_TRACE_RESOURCE_URI_MAPPING`      | `null`      | CSV of URL-to-resource-name mapping rules; e.g., `/foo/*,/bar/$*/baz`; [see "Custom URL-To-Resource Mapping"][16] |
+| `DD_TRACE_URL_AS_RESOURCE_NAMES_ENABLED` | `false` | Enable URL's as resource names; [see "Map Resource Names To Normalized URI"][17] |
 | `DD_<INTEGRATION>_ANALYTICS_ENABLED` | `false`     | Flag to enable trace analytics for relevant spans in a specific integration |
 
+#### Map Resource Names To Normalized URI
+
+<div class="alert alert-warning">
+This functionality is in public beta. Request early access by contacting <a href="/help">Datadog Support</a>.
```
Comment (priyanshi-gupta) [inline content/en/tracing/setup/php.md]: There is no "early access" required that they need to request. Can we only say "This functionality is in public beta. Contact support if more details are required."

Comment (apigirl) [inline content/en/tracing/setup/php.md]:
```suggestion
This functionality is in public beta. For more information, contact <a href="/help">Datadog Support</a>. 
```

**Pull request merged**